lib/utils.js

```diff
@@ -58,7 +58,7 @@ exports.forEach = function(arr, fn, scope){
 exports.map = function(arr, fn, scope){
   var result = [];
   for (var i = 0, l = arr.length; i < l; i++)
-    result.push(fn.call(scope, arr[i], i));
+    result.push(fn.call(scope, arr[i], i, arr));
```

Comment:
@travisjeffery, @boneskull because `.map()` used as [IE8 polyfill](https://github.com/mochajs/mocha/blob/master/lib/utils.js#L50), It's make sense to align it with the [spec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).
i.e: callback should call with 3 arguments: elmement, index and the array was called upon.
Thx.